aten/src/ATen/native/Fill.cpp

```diff
@@ -10,7 +10,26 @@ namespace native {
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ fill ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+namespace {
+  template <typename scalar_t>
+  inline void fill_fast(Tensor& self, Scalar value_scalar) {
+    auto value = value_scalar.to<scalar_t>();
+    scalar_t * dptr = reinterpret_cast<scalar_t *>(self.data_ptr());
+    *dptr = value;
+  }
+} // namspace
+
 Tensor& fill_out(Tensor& self, Scalar value) {
+  // When filling a number to 1-element CPU tensor, we want to skip
+  // everything but manipulate data ptr directly.
+  // Ideally this fast pass should be implemented in TensorIterator,
+  // but we also want to skip compute_types which in not avoidable
+  // in TensorIterator for now.
+  if (self.device() == at::kCPU && self.numel() == 1) {
+     AT_DISPATCH_ALL_TYPES(self.scalar_type(), "fill_out", [&]() {
```

Comment:
Test failures reminds me that all types are not really **all** types. I'm experimenting locally to see which is the best macro to use. But the rest of the PR is ready for review. ;)

Reply:
The rest of the PR looks good :-)